LuckyDragon/app/src/main/java/com/example/luckydragon/ProfileActivity.java

```diff
@@ -41,24 +28,51 @@ enum Mode {
         ORGANIZER,
         ADMIN,
     }
-
     private User user;
+    private GlobalApp.ROLE role;
+    private ProfileView profileView;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
         getSupportActionBar().setTitle("Profile");
 
-        // Unpack intent
-        Intent intent = getIntent();
+        // Get user from global app
         user = ((GlobalApp) getApplication()).getUser();
-        String role = intent.getStringExtra("role");
+        role = ((GlobalApp) getApplication()).getRole();
+
+        // Create profile view
+        profileView = new ProfileView(user, this);
+
+        // If user exists, update view
+        if(user != null) {
+            user.notifyObservers();
+        }
+    }
 
+    public void initializeView() {
         // Set profile info views
         TextView nameView = findViewById(R.id.nameTextView);
         TextView emailView = findViewById(R.id.emailTextView);
         TextView phoneNumberView = findViewById(R.id.phoneNumberTextView);
+
+        // Set name
+        nameView.setText(user.getName());
+        // Set email if it exists, otherwise hide the textview
+        if(user.getEmail() != null) {
+            emailView.setText(user.getEmail());
+        } else {
+            emailView.setVisibility(View.GONE);
+        }
+        // Set phone number if it exists, otherwise hide the textview
+        if(user.getPhoneNumber() != null) {
+            phoneNumberView.setText(user.getPhoneNumber());
+        } else {
+            Log.e("USER", "no phone number");
+            phoneNumberView.setVisibility(View.GONE);
```

Comment:
Put the logic for this in the view. The least amount of ui related code as necessary should be in the activity.

Reply:
Tried to do this but ran into some trouble. I have a fix but not sure if it is ideal. Take a look at onCreate() in my updated ProfileActivity and let me know what you think. I added comments to explain.